src/pages/api/gateway/[gatewayUID]/sensor/[macAddress]/config.ts

```diff
@@ -1,95 +1,95 @@
-  const { gatewayUID, macAddress } = req.query;
-  // Body params
-  const { loc, name } = req.body as NoteSensorConfigBody;
-  // Notehub values
-  const { hubBaseURL, hubAuthToken, hubProjectUID } = config;
-  // API path
-  const endpoint = `${hubBaseURL}/req?project=${hubProjectUID}&device=${gatewayUID}`;
-  // API headers
-  const headers = {
-    [HTTP_HEADER.CONTENT_TYPE]: HTTP_HEADER.CONTENT_TYPE_JSON,
-    [HTTP_HEADER.SESSION_TOKEN]: hubAuthToken,
-  };
+  return { gatewayUID, macAddress, location, name };
+}
 
-  // note.update payload
-  const noteUpdate = {
-    req: "note.update",
-    file: "config.db",
-    note: macAddress,
-    body: {
-      loc,
-      name,
-    },
-  };
+async function performRequest({
+  macAddress,
+  gatewayUID,
+  location,
+  name,
+}: ValidRequest) {
+  const app = services().getAppService();
+  try {
+    if (location) await app.setSensorLocation(gatewayUID, macAddress, location);
+    if (name) await app.setSensorName(gatewayUID, macAddress, name);
+  } catch (cause) {
+    throw new ErrorWithCause("Could not perform request", { cause });
+  }
+}
 
-  // Costruct body based on HTTP method
-  let postBody;
-  switch (req.method) {
-    case "POST":
-      // Check config body params
-      if (loc === undefined || name === undefined) {
-        res.status(400).json({ err: HTTP_STATUS.INVALID_CONFIG_BODY });
-        return;
-      }
-      postBody = noteUpdate;
-      break;
-    default:
-      // Other methods not allowed at this route
-      res.status(405).json({ err: HTTP_STATUS.METHOD_NOT_ALLOWED });
-      return;
+export default async function sensorConfigHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (!validateMethod(req, res)) {
+    return;
+  }
+  const validRequest = validateRequest(req, res);
+  if (!validRequest) {
+    return;
   }
 
   try {
-    // API call
-    const response: AxiosResponse = await axios.post(endpoint, postBody, {
-      headers,
-    });
-    // This call responds with a 200/OK even if the note was 404/NotFound.
-    // But it will contain an err property we can watch out for.
-    if ("err" in response.data) {
-      // Assert error type
-      const { err } = response.data as NotehubErr;
-      // Check the error message
-      if (err.includes("note-noexist")) {
-        // Return 204 error (request succeeded, but nothing to see here)
-        res.status(204).end();
-        return;
-      }
-      if (err.includes("insufficient permissions")) {
-        res.status(403).json({ err: HTTP_STATUS.UNAUTHORIZED });
-        return;
-      }
-    } else {
-      // Return JSON
-      res.status(200).json(response.data);
-      return;
-    }
-  } catch (err) {
-    // Return 500 error
-    res.status(500).json({ err: HTTP_STATUS.INTERNAL_ERR_CONFIG });
+    await performRequest(validRequest);
+    res.status(StatusCodes.OK).json({});
+  } catch (cause) {
+    throw new ErrorWithCause("could not handle sensor config", { cause });
```

Comment:
Nit pick - mixing returning a http status and throwing an error? Yeah, no real harm, but seems a bit strange.  I like the ErrorWithCause.

Reply:
I think I was afraid of losing the underlying cause of the error. So we really should add something that logs errors like this to the server log. (maybe even the frontend - optionally since that seems like a bit of a security bug - never know what's going to be in an error message.)